File: jupyter2/problems/function/projectile_height/go/projectile_height.go
package main
import "fmt"
import "math"

/** begin my answer */
func height(h0, v0, t float64) float64 {
    return h0 + v0*t - 0.5*9.8*t*t
}
/** end my answer */

func main() {
    if !(math.Abs(height(10,  1, 1) - 6.1) < 1e-5) { panic("wrong") }
    if !(math.Abs(height(20,  0, 2) - 0.4) < 1e-5) { panic("wrong") }
    if !(math.Abs(height(30, -1, 3) - -17.1) < 1e-5) { panic("wrong") }
    fmt.Println("OK")
}

File: jupyter2/problems/function/projectile_height/ml/projectile_height.ml
(** begin my answer *)

(** end my answer *)

let main () =
  assert (abs_float (height 10.0  1.0   1.0 -.   6.1) < 1e-5);
  assert (abs_float (height 20.0  0.0   2.0 -.   0.4) < 1e-5);
  assert (abs_float (height 30.0 (-1.0) 3.0 -. -17.1) < 1e-5);
  Printf.printf "OK\n"
;;

main()
